packages/blade/src/components/Card/CardHeader.tsx

```diff
@@ -0,0 +1,194 @@
+import Box from '~components/Box';
+import type { TextProps, TextVariant } from '~components/Typography';
+import { Heading, Text } from '~components/Typography';
+import type { IconComponent } from '~components/Icons';
+import type { CounterProps } from '~components/Counter';
+import { Counter } from '~components/Counter';
+import type { BadgeProps } from '~components/Badge';
+import { Badge } from '~components/Badge';
+import type { LinkProps } from '~components/Link';
+import { Link } from '~components/Link';
+import type { ButtonProps } from '~components/Button';
+import { Button } from '~components/Button';
+import { minHeight } from '~components/Button/BaseButton/buttonTokens';
+import { makeSpace } from '~utils';
+
+type CardHeaderProps = {
+  children?: React.ReactNode;
+};
+
+const Components = {
+  CardHeaderIcon: 'CardHeaderIcon',
+  CardHeaderCounter: 'CardHeaderCounter',
+  CardHeaderBadge: 'CardHeaderBadge',
+  CardHeaderText: 'CardHeaderText',
+  CardHeaderLink: 'CardHeaderLink',
+  CardHeaderIconButton: 'CardHeaderIconButton',
+};
+
+const CardHeaderIcon = ({ icon: Icon }: { icon: IconComponent }): React.ReactElement => {
+  useInsideCard('CardHeaderIcon');
+
+  return <Icon color="surface.text.normal.lowContrast" size="xlarge" />;
+};
+CardHeaderIcon.componentId = Components.CardHeaderIcon;
+
+const CardHeaderCounter = (props: CounterProps): React.ReactElement => {
+  useInsideCard('CardHeaderCounter');
+
+  return <Counter {...props} />;
+};
+CardHeaderCounter.componentId = Components.CardHeaderCounter;
+
+const CardHeaderBadge = (props: BadgeProps): React.ReactElement => {
+  useInsideCard('CardHeaderBadge');
+
+  return <Badge {...props} />;
+};
+CardHeaderBadge.componentId = Components.CardHeaderBadge;
+
+const CardHeaderText = (props: TextProps<{ variant: TextVariant }>): React.ReactElement => {
+  useInsideCard('CardHeaderText');
+
+  return <Text {...props} />;
+};
+CardHeaderText.componentId = Components.CardHeaderText;
+
+const CardHeaderLink = (props: LinkProps): React.ReactElement => {
+  useInsideCard('CardHeaderLink');
+
+  return <Link {...props} />;
+};
+CardHeaderLink.componentId = Components.CardHeaderLink;
+
+type CardHeaderIconButtonProps = Omit<
+  ButtonProps,
+  'variant' | 'size' | 'iconPosition' | 'isFullWidth' | 'children'
+> & {
+  icon: IconComponent;
+};
+
+const CardHeaderIconButton = (props: CardHeaderIconButtonProps): React.ReactElement => {
+  useInsideCard('CardHeaderIconButton');
+
+  return (
+    <Box width={makeSpace(minHeight.xsmall)}>
+      <Button {...props} variant="tertiary" size="xsmall" iconPosition="left" isFullWidth />
+    </Box>
+  );
+};
+CardHeaderIconButton.componentId = Components.CardHeaderIconButton;
+
+const getComponentId = (comp: React.ReactNode): string | null => {
+  if (!React.isValidElement(comp)) return null;
+  return (comp.type as any)?.componentId;
+};
+
+const isOfComponentId = (comp: React.ReactNode, id: string): boolean => {
```

Comment:
```suggestion
const isValidAllowedChildren = (component: React.ReactNode, id: string): boolean => {
```

Reply:
also move this to shared utils because @saurabhdaware is gonna need it for dropdown too?

Reply:
Will do.

Reply:
Done.